site/app/controllers/ChatroomController.php

```diff
@@ -0,0 +1,234 @@
+        $chatroom->setHostId($this->core->getUser()->getId());
+        $chatroom->setHostName($this->core->getUser()->getDisplayFullName());
+
+        $em->persist($chatroom);
+        $em->flush();
+
+        $this->core->addSuccessMessage("Chatroom successfully added");
+        return new RedirectResponse($this->core->buildCourseUrl(['chat']));
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat/{chatroom_id}", methods={"GET"})
+     * @param string $chatroom_id
+     * @return RedirectResponse|WebResponse
+     */
+     public function getChatroom(string $chatroom_id): WebResponse|RedirectResponse {
+         if (!is_numeric($chatroom_id)) {
+            $this->core->addErrorMessage("Invalid Chatroom ID");
+            return new RedirectResponse($this->core->buildCourseUrl(['chat']));
+        }
+
+        $repo = $this->core->getCourseEntityManager()->getRepository(Chatroom::class);
+        $chatroom = $repo->find($chatroom_id);
+
+        return new WebResponse(
+            'Chatroom',
+            'showChatroom',
+            $chatroom,
+        );
+     }
+
+     /**
+     * @Route("/courses/{_semester}/{_course}/chat/{chatroom_id}/anonymous", methods={"GET"})
+     * @param string $chatroom_id
+     * @return RedirectResponse|WebResponse
+     */
+     public function getChatroomAnon(string $chatroom_id): WebResponse|RedirectResponse {
+         if (!is_numeric($chatroom_id)) {
+            $this->core->addErrorMessage("Invalid Chatroom ID");
+            return new RedirectResponse($this->core->buildCourseUrl(['chat']));
+        }
+
+        $repo = $this->core->getCourseEntityManager()->getRepository(Chatroom::class);
+        $chatroom = $repo->find($chatroom_id);
+
+        return new WebResponse(
+            'Chatroom',
+            'showChatroom',
+            $chatroom, true,
+        );
+     }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat/deleteChatroom", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    public function deleteChatroom(): JsonResponse {
+        $chatroom_id = intval($_POST['chatroom_id'] ?? -1);
+        $em = $this->core->getCourseEntityManager();
+
+        /** @var \app\repositories\chat\ChatroomRepository $repo */
+        $repo = $em->getRepository(Chatroom::class);
+
+        /** @var Chatroom|null */
+        $chatroom = $repo->find($chatroom_id);
+
+        if ($chatroom === null) {
+            return JsonResponse::getFailResponse('Invalid Chatroom ID');
+        }
+
+        foreach ($chatroom->getMessages() as $message) {
+            $em->remove($message);
+        }
+        $em->remove($chatroom);
+        $em->flush();
+
+        return JsonResponse::getSuccessResponse();
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat/{chatroom_id}/edit", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    public function editChatroom(int $chatroom_id): RedirectResponse {
```

Comment:
Will the `int` type hint work here?  You used `string` earlier in this file, and I seem to recall that the type is always `string` for query parameters in Submitty.